src/imagej/__init__.py

```diff
@@ -1282,9 +1282,13 @@ def when_imagej_starts(f) -> None:
         as its sole argument, and called as the final action of the
         init function before it returns or blocks.
     """
-    # Add function to the list of callbacks to invoke upon start_jvm().
-    global _init_callbacks
-    _init_callbacks.append(f)
+    if sj.jvm_started():
```

Comment:
This doesn't seem like the right check, does it? Presumably, users would want to utilize an initialized ImageJ here. Is that always the case if the JVM is running? I'm assuming there's some period in between...

Reply:
Good catch! I lifted this from `scyjava` but thinking more deeply about it, the right check should be the existence of the `gateway` itself. So the check becomes:

```python
global gateway
if gateway:
    f(gateway)
```

I'll need to test this...

Reply:
Since we agreed that we will only have a `gateway` attribute on the `imagej` module created in GUI mode, this solution no longer works/makes sense. If this method works as intended when called before initializing ImageJ in GUI mode. However, if ImageJ is already initialized and the main thread consumed by the event loop then it becomes impossible to call `when_imagej_starts_()`, making any check of a pre-existing `gateway` attribute pointless. As I see it, there is no good way to detect if an ImageJ in addition to obtaining a handle on the `gateway` to use with the desired callback. Thus, I decided to remove this commit and revert it back to just loading the callbacks into the `_init_callbacks` global list.

Reply:
Yeah, I think that makes sense too.